pkg/pillar/scripts/device-steps.sh

```diff
@@ -203,7 +204,7 @@ if [ -c $TPM_DEVICE_PATH ] && ! [ -f $CONFIGDIR/disable-tpm ] && [ "$P3_FS_TYPE"
     #It is a device with TPM, and formatted with ext4, setup fscrypt
     echo "$(date -Ins -u) EXT4 partitioned $PERSISTDIR, enabling fscrypt"
     #Initialize fscrypt algorithm, hash length etc.
-    $BINDIR/vaultmgr -c "$CURPART" setupVaults
+    LOG_TO_SYSLOG=1 $BINDIR/vaultmgr -c "$CURPART" setupVaults
```

Comment:
Is there any reason not to set LOG_TO_SYSLOG as a environment variable at the top of the script? My cursory scanning of device-steps shows that either all services log to syslog or not -- so why prefix every single one with LOG_TO_SYSLOG=1 ?

Reply:
@rvs I tried doing that by:
1) putting "export LOG_TO_SYSLOG=1"
2) putting "LOG_TO_SYSLOG=1"
at the top of device-steps. I have noticed the individual services were not picking up the env variable. Did I miss setting something? Or should it be done differently?

Reply:
Given that we're using POSIX sh, can you please try the following diff:
```
diff --git a/pkg/pillar/scripts/device-steps.sh b/pkg/pillar/scripts/device-steps.sh
index 81e95030..d5e0dd42 100755
--- a/pkg/pillar/scripts/device-steps.sh
+++ b/pkg/pillar/scripts/device-steps.sh
@@ -3,6 +3,7 @@
 # Copyright (c) 2018 Zededa, Inc.
 # SPDX-License-Identifier: Apache-2.0

+LOG_TO_SYSLOG=1
 USE_HW_WATCHDOG=1
 CONFIGDIR=/config
 PERSISTDIR=/persist
@@ -23,6 +24,8 @@ TPM_DEVICE_PATH="/dev/tpmrm0"

 PATH=$BINDIR:$PATH

+export LOG_TO_SYSLOG
+
 echo "$(date -Ins -u) Starting device-steps.sh"
 echo "$(date -Ins -u) EVE version: $(cat $BINDIR/versioninfo)"
```

Reply:
I'll do that.

Reply:
It worked @rvs . I have committed the changes.